src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -112,21 +114,209 @@ private static IDictionary GetEnvironmentVariablesCore(EnvironmentVariableTarget
+
+                case SpecialFolder.MyMusic:
+                    return IsMac ? Path.Combine(home, "Music") : ReadXdgDirectory(home, "XDG_MUSIC_DIR", "Music");
+                case SpecialFolder.MyPictures:
+                    return IsMac ? Path.Combine(home, "Pictures") : ReadXdgDirectory(home, "XDG_PICTURES_DIR", "Pictures");
+                case SpecialFolder.Fonts:
+                    return IsMac ? Path.Combine(home, "Library", "Fonts") : Path.Combine(home, ".fonts");
+
+                case SpecialFolder.Favorites:
+                    if (IsMac) return Path.Combine(home, "Library", "Favorites");
+                    break;
+                case SpecialFolder.InternetCache:
+                    if (IsMac) return Path.Combine(home, "Library", "Caches");
                     break;
+            }
+
+            // No known path for the SpecialFolder
+            return string.Empty;
+        }
+
+        private static string GetXdgConfig(string home)
+        {
+            // "$XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored."
+            // "If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used."
+            string config = GetEnvironmentVariable("XDG_CONFIG_HOME");
+            if (string.IsNullOrEmpty(config))
+            {
+                config = Path.Combine(home, ".config");
+            }
+            return config;
+        }
+
+        private static string ReadXdgDirectory(string homeDir, string key, string fallback)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(homeDir), $"Expected non-empty homeDir");
+            Debug.Assert(!string.IsNullOrEmpty(key), $"Expected non-empty key");
+            Debug.Assert(!string.IsNullOrEmpty(fallback), $"Expected non-empty fallback");
 
-                // TODO: Add more special folder handling
+            string envPath = GetEnvironmentVariable(key);
+            if (!string.IsNullOrEmpty(envPath))
+            {
+                return envPath;
+            }
+
+            string userDirsPath = Path.Combine(GetXdgConfig(homeDir), "user-dirs.dirs");
+            if (Interop.Sys.Access(userDirsPath, Interop.Sys.AccessMode.R_OK) == 0)
+            {
+                try
+                {
+                    // TODO #11151: Replace with direct usage of File.ReadLines or equivalent once we have access to System.IO.FileSystem here.
+                    Type fileType = Type.GetType("System.IO.File, System.IO.FileSystem, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+                    IEnumerable<string> lines = fileType?.GetTypeInfo().GetDeclaredMethod("ReadLines").Invoke(null, new object[] { userDirsPath }) as IEnumerable<string>;
+                    if (lines != null)
+                    {
+                        foreach (string line in lines)
```

Comment:
I was looking for the specification for this file, and ran across https://www.freedesktop.org/wiki/Software/xdg-user-dirs/.

> They can also set or change the default directories and their initial values in /etc/xdg/user-dirs.defaults.

I'm not sure how much of this we want to conform to. I'm just observing, and wondering where the right balance is between "never-ending feature" and "fails to meet user expectations".

I don't suppose any of this comes from a reliably-available library?